fdbserver/DataDistributionQueue.actor.cpp

```diff
@@ -41,20 +41,97 @@
+	       reason == DataMovementReason::REBALANCE_OVERUTILIZED_TEAM;
+}
+
 inline bool isMountainChopperPriority(int priority) {
 	return priority == SERVER_KNOBS->PRIORITY_REBALANCE_OVERUTILIZED_TEAM ||
 	       priority == SERVER_KNOBS->PRIORITY_REBALANCE_READ_OVERUTIL_TEAM;
 }
+
+inline bool isMountainChopperPriority(DataMovementReason reason) {
+	return reason == DataMovementReason::REBALANCE_OVERUTILIZED_TEAM ||
+	       reason == DataMovementReason::REBALANCE_READ_OVERUTIL_TEAM;
+}
+
 inline bool isValleyFillerPriority(int priority) {
 	return priority == SERVER_KNOBS->PRIORITY_REBALANCE_UNDERUTILIZED_TEAM ||
 	       priority == SERVER_KNOBS->PRIORITY_REBALANCE_READ_UNDERUTIL_TEAM;
 }
 
+inline bool isValleyFillerPriority(DataMovementReason reason) {
+	return reason == DataMovementReason::REBALANCE_UNDERUTILIZED_TEAM ||
+	       reason == DataMovementReason::REBALANCE_READ_UNDERUTIL_TEAM;
+}
+
+int mapMovementReasonToKnobPriority(DataMovementReason reason) {
+	int priority;
+	switch (reason) {
+	case DataMovementReason::RECOVER_MOVE:
+		priority = SERVER_KNOBS->PRIORITY_RECOVER_MOVE;
+		break;
+	case DataMovementReason::REBALANCE_UNDERUTILIZED_TEAM:
+		priority = SERVER_KNOBS->PRIORITY_REBALANCE_UNDERUTILIZED_TEAM;
+		break;
+	case DataMovementReason::REBALANCE_OVERUTILIZED_TEAM:
+		priority = SERVER_KNOBS->PRIORITY_REBALANCE_OVERUTILIZED_TEAM;
+		break;
+	case DataMovementReason::REBALANCE_READ_OVERUTIL_TEAM:
+		priority = SERVER_KNOBS->PRIORITY_REBALANCE_READ_OVERUTIL_TEAM;
+		break;
+	case DataMovementReason::REBALANCE_READ_UNDERUTIL_TEAM:
+		priority = SERVER_KNOBS->PRIORITY_REBALANCE_READ_UNDERUTIL_TEAM;
+		break;
+	case DataMovementReason::PERPETUAL_STORAGE_WIGGLE:
+		priority = SERVER_KNOBS->PRIORITY_PERPETUAL_STORAGE_WIGGLE;
+		break;
+	case DataMovementReason::TEAM_HEALTHY:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_HEALTHY;
+		break;
+	case DataMovementReason::TEAM_CONTAINS_UNDESIRED_SERVER:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_CONTAINS_UNDESIRED_SERVER;
+		break;
+	case DataMovementReason::TEAM_REDUNDANT:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_REDUNDANT;
+		break;
+	case DataMovementReason::MERGE_SHARD:
+		priority = SERVER_KNOBS->PRIORITY_MERGE_SHARD;
+		break;
+	case DataMovementReason::POPULATE_REGION:
+		priority = SERVER_KNOBS->PRIORITY_POPULATE_REGION;
+		break;
+	case DataMovementReason::TEAM_UNHEALTHY:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_UNHEALTHY;
+		break;
+	case DataMovementReason::TEAM_2_LEFT:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_2_LEFT;
+		break;
+	case DataMovementReason::TEAM_1_LEFT:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_1_LEFT;
+		break;
+	case DataMovementReason::TEAM_FAILED:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_FAILED;
+		break;
+	case DataMovementReason::TEAM_0_LEFT:
+		priority = SERVER_KNOBS->PRIORITY_TEAM_0_LEFT;
+		break;
+	case DataMovementReason::SPLIT_SHARD:
+		priority = SERVER_KNOBS->PRIORITY_SPLIT_SHARD;
+		break;
+	default:
```

Comment:
Best to remove the default; then the compiler will complain when there is an unhandled case if new ones are added in the future.

Reply:
fix it by removing default